### PR TITLE
Preapps 1179 - First Draft autosave collapses CC/Bcc, moves cursor focus

### PR DESCRIPTION
**Pull request by itsmeprashant**

### CODE REVIEW

File: src/schema/schema.graphql [hunk 1]
```diff
@@ -1406,9 +1406,14 @@ input CalendarItemMessageInput {
...
 }
 
+input NewAttachmentInput {
+	attachmentId: ID
+}
+
 input SendMessageInput {
 	id: ID #id
 	origId: ID #origid
+	attach: [NewAttachmentInput]
```
Comment (silentsakky) [inline src/schema/schema.graphql]: we already have AttachmentInput for this

File: src/utils/normalize-mime-parts.ts [hunk 2]
```diff
@@ -89,6 +89,12 @@ export function normalizeMimeParts(
 				part.contentType !== 'application/pkcs7-mime' &&
 					part.contentType !== 'application/pkcs7-signature' &&
 					(acc[mode] || (acc[mode] = [])).push(processAttachment(part));
+
+				message.attributes = message.attributes || {};
+				message.attributes.isEncrypted =
+					part.contentType === 'application/pkcs7-mime';
+				message.attributes.isSigned =
+					part.contentType === 'application/pkcs7-signature';
```
Comment (silentsakky) [inline src/utils/normalize-mime-parts.ts]: also add application/x-pkcs7-signature check

File: src/utils/normalize-mime-parts.ts [hunk 3]
```diff
@@ -89,6 +89,12 @@ export function normalizeMimeParts(
 				part.contentType !== 'application/pkcs7-mime' &&
 					part.contentType !== 'application/pkcs7-signature' &&
 					(acc[mode] || (acc[mode] = [])).push(processAttachment(part));
+
+				message.attributes = message.attributes || {};
+				message.attributes.isEncrypted =
+					part.contentType === 'application/pkcs7-mime';
```
Comment (silentsakky) [inline src/utils/normalize-mime-parts.ts]: do we need check for multipart/signed also?

Comment (itsmeprashant) [reply src/utils/normalize-mime-parts.ts]: Yes, we will add that when we gonna support Outlook/Thunderbird client.

File: src/utils/normalize-mime-parts.ts (hunk 2 above)
Comment (itsmeprashant) [reply src/utils/normalize-mime-parts.ts]: sure.

File: src/schema/schema.graphql (hunk 1 above)
Comment (itsmeprashant) [reply src/schema/schema.graphql]: That is different and used for other purpose. We can't use that.

Comment (silentsakky) [reply src/schema/schema.graphql]: I think it's same, it has extra key which can be ignored

**Pull request merged**